File: app/src/main/java/com/wp/csmu/classschedule/view/adapter/ScoreRecyclerAdapter.java
package com.wp.csmu.classschedule.view.adapter;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.wp.csmu.classschedule.R;
import com.wp.csmu.classschedule.view.bean.Score;

import java.util.List;

public class ScoreRecyclerAdapter extends RecyclerView.Adapter<ScoreRecyclerAdapter.ViewHolder> {
    private List<Score>data;
    private OnClickListener listener;

    public ScoreRecyclerAdapter(List<Score>data,@NonNull OnClickListener listener) {
        this.data = data;
        this.listener=listener;
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view=LayoutInflater.from(parent.getContext()).inflate(R.layout.score_recyclerview_item,parent,false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(final ViewHolder holder, int position) {
        holder.subjectName.setText(data.get(position).getSubject());
        holder.subjectEnglishName.setText(data.get(position).getSubjectEnglish());
        holder.score.setText(String.valueOf(data.get(position).getScore()));
        holder.itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                listener.onClick(v,holder.getAdapterPosition());
            }
        });
    }

    @Override
    public int getItemCount() {
        return data==null?0:data.size();
    }

    public void updateData(List<Score>newData){
        this.data=newData;
        notifyDataSetChanged();
    }

    static class ViewHolder extends RecyclerView.ViewHolder{
        TextView subjectName;
        TextView subjectEnglishName;
        TextView score;
        ViewHolder(View itemView) {
            super(itemView);
            subjectName=itemView.findViewById(R.id.scoreRecyclerItemTvSubject);
            subjectEnglishName=itemView.findViewById(R.id.scoreRecyclerItemTvSubjectEnglish);
            score=itemView.findViewById(R.id.scoreRecyclerItemTvScore);
        }
    }

    public interface OnClickListener{
        void onClick(View view, int position);
    }
}

File: app/src/main/java/com/wp/csmu/classschedule/view/adapter/ScoreRecyclerAdapter.kt
package com.wp.csmu.classschedule.view.adapter

import android.graphics.Color
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.wp.csmu.classschedule.R
import com.wp.csmu.classschedule.view.bean.Score

class ScoreRecyclerAdapter(private var data: List<Score>?, private val listener: OnClickListener) : RecyclerView.Adapter<ScoreRecyclerAdapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.score_recyclerview_item, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.subjectName.text = data!![position].name
        holder.score.text = data!![position].score
        try {
            if (data!![position].score.toDouble() < 60.0) holder.score.setTextColor(Color.RED)
        } catch (e: Exception) {
        }
        holder.itemView.setOnClickListener { v -> listener.onClick(v, holder.adapterPosition) }
    }

    override fun getItemCount(): Int {
        return if (data == null) 0 else data!!.size
    }

    fun updateData(newData: List<Score>) {
        this.data = newData
        notifyDataSetChanged()
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var subjectName: TextView = itemView.findViewById(R.id.scoreRecyclerItemTvSubject)
        var score: TextView = itemView.findViewById(R.id.scoreRecyclerItemTvScore)
    }

    interface OnClickListener {
        fun onClick(view: View, position: Int)
    }
}
